datastore/repository.go

```diff
@@ -49,6 +49,7 @@ type GroupRepository interface {
 type ApplicationRepository interface {
 	CreateApplication(context.Context, *Application) error
 	LoadApplicationsPaged(context.Context, string, string, Pageable) ([]Application, PaginationData, error)
+	IsAppTitleUnique(context.Context, string, string) (bool, error)
```

Comment:
@danvixent Hmm. Why isn't this just `IsAppTitleUnique(context.Context, string, string) error`. This simplifies the API and the test where we are checking for `error` and the `bool` value.

Reply:
I agree that the definition is awkward, and i personally don't like it, but in this case, i think it is necessary, we are trying to check if an app name is unique => `true` or `false`, `error` reports any error we face along the way.

Reply:
@Subomi